app/react/Forms/components/MultiSelect.tsx

```diff
@@ -83,26 +82,59 @@ abstract class MultiSelectBase<ValueType> extends Component<
     return [];
   }
 
-  private searchInputRef: RefObject<HTMLInputElement>;
+  abstract getCheckedList(): string[];
+
+  abstract markUnchecked(value: ValueType, option: Option): ValueType;
+
+  abstract markChecked(value: ValueType, option: Option): ValueType;
+
+  private getGroupStateValue(value: ValueType, group: Option) {
```

Comment:
Code Climate is complaining about the code repetition for:

getGroupStateValue
getOnStateValue
getOffStateValue

And it makes sense. It could be easily refactored. We can reduce the lines number in the file, which is good.
One suggestion is that the three methods could stay as they help to understand the code with the names, but the logic inside them could be extracted.